Zend/zend_vm_def.h

```diff
@@ -1620,14 +1628,17 @@ ZEND_VM_HELPER_EX(zend_fetch_var_address_helper, CONST|TMPVAR|CV, UNUSED|CONST|V
 	}
 
 ZEND_VM_C_LABEL(fetch_var_return):
-	ZEND_ASSERT(retval != NULL);
-	if (type == BP_VAR_R || type == BP_VAR_IS) {
-		if (/*type == BP_VAR_R &&*/ Z_ISREF_P(retval) && Z_REFCOUNT_P(retval) == 1) {
-			ZVAL_UNREF(retval);
-		}
-		ZVAL_COPY(EX_VAR(opline->result.var), retval);
+	if (!retval) {
```

Comment:
I'd prefer moving this check to only the cases where it can occur. In particular, this code is (in PHP 7.0) shared between static member fetch and non-CV variable fetches, while the NULL check is only relevant for static members. In particular, I would suggest to a) use `retval = EG(uninitialized_zval);` in the two places above where you currently have `retval = NULL;` and b) extend the NULL check at https://github.com/php/php-src/pull/2436/files#diff-3054389ad750ce9a9f5895cd6d27800fR1566 (i.e. rewrite it as `if (retval) { if (const) { cache } } else { retval = &EG(uninitialized_zval); }`).